main.go

```diff
@@ -149,13 +152,27 @@ func main() {
 
 	ctrl.SetLogger(log)
 
+	managerNamespace := ""
+	var newCache cache.NewCacheFunc
+
 	if autoScalingRunnerSetOnly {
 		// We don't support metrics for AutoRunnerScaleSet for now
 		metricsAddr = "0"
+
+		managerNamespace = os.Getenv("CONTROLLER_MANAGER_POD_NAMESPACE")
```

Comment:
I don't know if this would be a good idea, but I'll just ask for discussion:
Should we default to the `default` namespace if no namespace is provided, like k8s does? It seems more dangerous in case someone makes a mistake, but it also aligns well with the way k8s treats the empty value

Reply:
This env var gets its value from 

```
    - name: CONTROLLER_MANAGER_POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
```

so it depends on where arc is installed?

Reply:
Yes, but I am not sure if it is going to receive `""` if no namespace is set :relaxed: I'll play around to make sure, because I am not sure how it would look like if `metadata.namespace` is not set

Reply:
It should never be empty though, you can't create a resource outside a namespace?